### PR TITLE
Introduce IEndpointAndServiceDefinition

### PR DESCRIPTION
**Pull request by davidkallesen**

### CODE REVIEW

File: sample/test/Demo.Api.Contracts.Tests/Demo.Api.Contracts.Tests.csproj [hunk 1]
```diff
@@ -22,4 +22,8 @@
     <ProjectReference Include="..\..\src\Demo.Api\Demo.Api.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Atc.Test" Version="1.0.79" />
```
Comment (perkops) [inline sample/test/Demo.Api.Contracts.Tests/Demo.Api.Contracts.Tests.csproj]: build.props instead

File: sample/test/Demo.Api.IntegrationTests/Demo.Api.IntegrationTests.csproj [hunk 2]
```diff
@@ -23,4 +23,8 @@
     <ProjectReference Include="..\..\src\Demo.Api\Demo.Api.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Atc.Test" Version="1.0.79" />
```
Comment (perkops) [inline sample/test/Demo.Api.IntegrationTests/Demo.Api.IntegrationTests.csproj]: build.props instead

File: sample/test/Demo.Domain.Tests/Demo.Domain.Tests.csproj [hunk 3]
```diff
@@ -22,4 +22,8 @@
     <ProjectReference Include="..\..\src\Demo.Domain\Demo.Domain.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Atc.Test" Version="1.0.79" />
```
Comment (perkops) [inline sample/test/Demo.Domain.Tests/Demo.Domain.Tests.csproj]: build.props instead

File: test/Atc.Rest.MinimalApi.Tests/Atc.Rest.MinimalApi.Tests.csproj [hunk 4]
```diff
@@ -25,4 +25,8 @@
     <ProjectReference Include="..\..\src\Atc.Rest.MinimalApi\Atc.Rest.MinimalApi.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Atc.Test" Version="1.0.79" />
```
Comment (perkops) [inline test/Atc.Rest.MinimalApi.Tests/Atc.Rest.MinimalApi.Tests.csproj]: build.props instead

File: sample/.editorconfig [hunk 5]
```diff
@@ -517,4 +517,6 @@ dotnet_diagnostic.CA2016.severity = none            # Forward the CancellationTo
 dotnet_diagnostic.MA0004.severity = none            # Consider calling ConfigureAwait.
 dotnet_diagnostic.MA0040.severity = none            # Forward the CancellationToken parameter to methods that take one - False-Positive
 
+dotnet_diagnostic.SA1402.severity = none            # File may only contains a single type - not relevant for multi records in single file by design
```
Comment (perkops) [inline sample/.editorconfig:520]: Is this suppression still needed?

File: src/Atc.Rest.MinimalApi/Extensions/EndpointDefinitionExtensions.cs [hunk 6]
```diff
@@ -33,9 +33,46 @@ public static class EndpointDefinitionExtensions
         services.AddSingleton(endpointDefinitions as IReadOnlyCollection<IEndpointDefinition>);
     }
 
+    /// <summary>
+    /// Adds the endpoint definitions to the specified service collection by scanning the assemblies of the provided marker types.
+    /// This method looks for types that implement the <see cref="IEndpointDefinition"/> interface and are neither abstract nor an interface,
```
Comment (perkops) [inline src/Atc.Rest.MinimalApi/Extensions/EndpointDefinitionExtensions.cs]: and IEndpointAndServiceDefinition interfaces (with cref on IEndpointAndServiceDefinition)

File: sample/test/Demo.Api.Contracts.Tests/Demo.Api.Contracts.Tests.csproj (hunk 1 above)
Comment (davidkallesen) [reply sample/test/Demo.Api.Contracts.Tests/Demo.Api.Contracts.Tests.csproj]: Check :-)

**Review by perkops — APPROVED** [review]

**Pull request merged**